src/org/intellij/erlang/completion/QuoteInsertHandler.java

```diff
@@ -52,24 +52,41 @@ public static void process(@NotNull Project project, @NotNull String name, @NotN
 
   public static void process(@NotNull Project project, @NotNull String name, @Nullable String moduleName, @NotNull InsertionContext context) {
     if (needQuotation(project, moduleName)) {
-      Editor editor = context.getEditor();
-      Document document = editor.getDocument();
-      context.commitDocument();
       int startOffset = context.getStartOffset();
-      document.insertString(startOffset, QUOTA);
-      document.insertString(startOffset + moduleName.length() + 1, QUOTA);
+      insertQuotesIfAbsent(context, startOffset, startOffset + moduleName.length());
     }
     if (needQuotation(project, name)) {
-      Editor editor = context.getEditor();
-      Document document = editor.getDocument();
-      context.commitDocument();
       int tailOffset = context.getTailOffset();
       int startOffset = moduleName == null ? context.getStartOffset() : tailOffset - name.length();
+      tailOffset += insertQuotesIfAbsent(context, startOffset, tailOffset);
+      context.getEditor().getCaretModel().moveToOffset(tailOffset);
+      context.setTailOffset(tailOffset);
+    }
+  }
+
+  private static int insertQuotesIfAbsent(@NotNull InsertionContext context, int startOffset, int tailOffset) {
+    int insertedQuotesCount = 0;
+    Editor editor = context.getEditor();
+    Document document = editor.getDocument();
+    context.commitDocument();
+
+    if (!alreadyHasQuote(context, startOffset - 1)) {
       document.insertString(startOffset, QUOTA);
-      document.insertString(tailOffset + 1, QUOTA);
-      editor.getCaretModel().moveToOffset(tailOffset + 2);
-      context.setTailOffset(tailOffset + 2);
+      context.commitDocument();
+      insertedQuotesCount++;
+      tailOffset++;
+    }
+    if (!alreadyHasQuote(context, tailOffset)) {
+      document.insertString(tailOffset, QUOTA);
+      insertedQuotesCount++;
     }
+    return  insertedQuotesCount;
+  }
+
+  private static boolean alreadyHasQuote(@NotNull InsertionContext context, int position) {
+    char[] file = context.getFile().textToCharArray();
```

Comment:
I have found only one alternative - `context.getFile().getText()`. And it also creates text copy, as I understood.
And I don't like both of these variants. May be you have some ideas how it would be better to be done?